### PR TITLE
Add PN532, MFRC522, PN5180 NFC readers, Mifare, Ultralight and NDEF

### PR DESCRIPTION
**Pull request by Ellerbach**
## Description

Add PN532, MFRC522, PN5180 NFC readers, Mifare, Ultralight and NDEF

## Motivation and Context

- Add 3 NFC readers:
  - PN532
  - PN5180
  - MFRC522
- Add Type A cards:
  - Mifare
  - Ultralight
-  NDEF
- Full documentation
- Full examples

## How Has This Been Tested?

- Each reader fully tested with Ultralight and Mifare card, including NDEF
- SPI, Serial and I2C tests except for MFRC522 (only SPI)

## Screenshots

Examples of output:
image

image

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [x] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/nanoFramework.IoT.Device.CodeConverter/appsettings.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
-  "SourceDirectory": "e:\\GitHub\\DotNet-iot\\src\\devices",
+  "SourceDirectory": "c:\\Repos\\iot\\src\\devices",
```
Comment (josesimoes) [inline src/nanoFramework.IoT.Device.CodeConverter/appsettings.json:2]: Everyone that touches this seems to be fighting to put it's local path! 🤣

**Review by josesimoes — COMMENTED** [review]
You seem to have put here a new batch of devices generated... could you please remove those from the PR, so it's easy to review?

It's OK to add those in another PR. 😉

File: src/nanoFramework.IoT.Device.CodeConverter/appsettings.json (hunk 1 above)
Comment (Ellerbach) [reply src/nanoFramework.IoT.Device.CodeConverter/appsettings.json:2]: :-D Does it really matter? :-)

**Comment by Ellerbach** [comment]
> You seem to have put here a new batch of devices generated... could you please remove those from the PR, so it's easy to review?
> 
> It's OK to add those in another PR. 😉

Well, I've done all at one... As I had to fix the generated list and as I had to clean the folder as well. As it's in one of my previous commit and I4ve been adjusting quite some elements, just focus on the specific folders:
- Card and anything under
- Pn532
- Pn5180
- Mfrc522
- anything under /devices with 1 file modification, the ListSomething one which I had updated

File: src/nanoFramework.IoT.Device.CodeConverter/appsettings.json (hunk 1 above)
Comment (josesimoes) [reply src/nanoFramework.IoT.Device.CodeConverter/appsettings.json:2]: No. Was just noticing that and found it amusing. 😉

**Pull request merged**